### PR TITLE
[OCPQE-13915]Add registry storage pvc for special vpshere cluster

### PR DESCRIPTION
**Pull request by wewang58**
When migrate some vpshere profiles to prow ci, found registry using pvc storage for some profiles, like as follow:
/versioned-installer-vmc7-disconnected_techpreview-thin_pvc-ci
./versioned-installer-vmc7-http_proxy-techpreview-external_lb-thin_pvc-ci
./versioned-installer-vmc7-ovn-winc-thin_pvc-ci
@jianlinliu suggest we an choose one profile to set registry storage for pvc, so choose profile installer-rehearse-vsphere-discon  to submit a pr

### CODE REVIEW

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-vsphere-ipi-zones-techpreview-p3-f28

**Comment by wewang58** [comment]
/pj-rehearse pull-ci-openshift-machine-api-operator-release-4.13-e2e-vsphere-ovn

**Comment by wewang58** [comment]
/pj-rehearse pull-ci-openshift-vsphere-problem-detector-release-4.12-e2e-vsphere

File: ci-operator/step-registry/cucushift/installer/rehearse/vsphere/ipi/disconnected/techpreview/provision/cucushift-installer-rehearse-vsphere-ipi-disconnected-techpreview-provision-chain.yaml [hunk 1]
```diff
@@ -19,5 +19,8 @@ chain:
   - name: BASTION_HOST_SUBNET
     default: "qe-segment"
     documentation: the subnet to provision bastion host
+  - name: "STEPUP_IMAGE_REGISTRY_WITH_PVC"
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/vsphere/ipi/disconnected/techpreview/provision/cucushift-installer-rehearse-vsphere-ipi-disconnected-techpreview-provision-chain.yaml:22]: Suggest to enable this option in a non-techpreview install workflow, e.g: `vsphere-ipi-disconnected`

File: ci-operator/step-registry/ipi/install/vsphere/registry/ipi-install-vsphere-registry-commands.sh [hunk 2]
```diff
@@ -5,10 +5,28 @@ set -o errexit
...
 
 export KUBECONFIG=${SHARED_DIR}/kubeconfig
-
-echo "$(date -u --rfc-3339=seconds) - Configuring image registry with emptyDir..."
-oc patch configs.imageregistry.operator.openshift.io cluster --type merge --patch '{"spec":{"managementState":"Managed","storage":{"emptyDir":{}}}}'
-
+#Create pvc for image registry storage
+if [[ "${STEPUP_IMAGE_REGISTRY_WITH_PVC}" == "yes" ]]; then
+  echo "$(date -u --rfc-3339=seconds) - Setting the image registry storage to use the manual setting storage..."
+cat >> "${SHARED_DIR}/registry-pvc.yaml" << EOF
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/install/vsphere/registry/ipi-install-vsphere-registry-commands.sh:11]: How about adding two more whitespaces for indent

File: ci-operator/step-registry/cucushift/installer/rehearse/vsphere/ipi/disconnected/techpreview/provision/cucushift-installer-rehearse-vsphere-ipi-disconnected-techpreview-provision-chain.yaml (hunk 1 above)
Comment (wewang58) [reply ci-operator/step-registry/cucushift/installer/rehearse/vsphere/ipi/disconnected/techpreview/provision/cucushift-installer-rehearse-vsphere-ipi-disconnected-techpreview-provision-chain.yaml]: @jianlinliu  thanks review, updated it according your comments.

**Comment by wewang58** [comment]
@xiuwang  Could you help to review it, thanks

**Comment by wewang58** [comment]
/pj-rehearse pull-ci-openshift-machine-api-operator-release-4.13-e2e-vsphere-ovn

**Comment by wewang58** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-vsphere-ipi-disconnected-p2-f14

File: ci-operator/step-registry/cucushift/installer/rehearse/vsphere/ipi/disconnected/provision/cucushift-installer-rehearse-vsphere-ipi-disconnected-provision-chain.yaml [hunk 3]
```diff
@@ -18,5 +18,8 @@ chain:
   - name: BASTION_HOST_SUBNET
     default: "qe-segment"
     documentation: the subnet to provision bastion host
+  - name: "STEPUP_IMAGE_REGISTRY_WITH_PVC"
```
Comment (xiuwang) [inline ci-operator/step-registry/cucushift/installer/rehearse/vsphere/ipi/disconnected/provision/cucushift-installer-rehearse-vsphere-ipi-disconnected-provision-chain.yaml:21]: @wewang58 STEPUP is your intended, or you mean SETUP?

File: ci-operator/step-registry/ipi/install/vsphere/registry/ipi-install-vsphere-registry-commands.sh [hunk 4]
```diff
@@ -5,10 +5,28 @@ set -o errexit
...
+metadata:
+  name: registry-pvc
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 10Gi
+EOF
+  oc patch config.image/cluster -p '{"spec":{"managementState":"Removed","storage":null}}' --type=merge &&
```
Comment (xiuwang) [inline ci-operator/step-registry/ipi/install/vsphere/registry/ipi-install-vsphere-registry-commands.sh:23]: This should be first time to setup registry, the default status is Removed already

File: ci-operator/step-registry/cucushift/installer/rehearse/vsphere/ipi/disconnected/provision/cucushift-installer-rehearse-vsphere-ipi-disconnected-provision-chain.yaml (hunk 3 above)
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/vsphere/ipi/disconnected/provision/cucushift-installer-rehearse-vsphere-ipi-disconnected-provision-chain.yaml:21]: Suggest to create a dedicated folder to save the workflow, create it under `vsphere/ipi/disconnected/image-registry-pvc/`

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/rehearse/vsphere/ipi/disconnected/provision/cucushift-installer-rehearse-vsphere-ipi-disconnected-provision-chain.yaml:21]: Just like what we did for fips: https://github.com/openshift/release/tree/master/ci-operator/step-registry/cucushift/installer/rehearse/vsphere/ipi/proxy/fips

Comment (wewang58) [reply ci-operator/step-registry/cucushift/installer/rehearse/vsphere/ipi/disconnected/provision/cucushift-installer-rehearse-vsphere-ipi-disconnected-provision-chain.yaml:21]: @jianlinliu  if it's ok that just add the workflow in 4.12 and 4.13 amd64 nightly files,  and set p2 and f7 ?

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/rehearse/vsphere/ipi/disconnected/provision/cucushift-installer-rehearse-vsphere-ipi-disconnected-provision-chain.yaml:21]: My suggestion is adding a new install workflow for this test scenario, while in our daily ci yam files, we replace all the occurrence of  `vsphere-ipi-disconnected` with this new workflow.

**Comment by wewang58** [comment]
@jianlinliu @xiuwang Updated it according to your comments thanks,seems failed jobs are not related to the pr.

**Pull request merged**